src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -106,15 +108,26 @@ public Dictionary(int capacity, IEqualityComparer<TKey>? comparer)
             AddRange(collection);
         }
 
-        private void AddRange(IEnumerable<KeyValuePair<TKey, TValue>> collection)
+        private void AddRange(IEnumerable<KeyValuePair<TKey, TValue>> enumerable)
         {
-            // It is likely that the passed-in dictionary is Dictionary<TKey,TValue>. When this is the case,
+            // Special-case ReadOnlyDictionary<TKey, TValue> to extract its underlying dictionary
```

Comment:
How common is it specifically for ReadOnlyDictionary to be passed to this method?

Reply:
It's not super common.  It would likely come about from a) consuming an IDictionary that's actually a ReadOnlyDictionary, and then b) wanting a copy of that so using that IDictionary to construct the Dictionary.  But, I can remove it if you don't think it's warranted.